mlrun/artifacts/plots.py

```diff
@@ -121,7 +121,56 @@ def get_body(self):
             self.options["title"] = self.title or self.key
         data = [self.header] + self.rows
         return (
-            chart_template.replace("$data$", dict_to_json(data))
+            self._TEMPLATE.replace("$data$", dict_to_json(data))
             .replace("$opts$", dict_to_json(self.options))
             .replace("$chart$", self.chart)
         )
+
+
+class BokehArtifact(Artifact):
+    """
+    Bokeh artifact is an artifact for saving Bokeh generated figures. They will be stored in html format.
+    """
+
+    kind = "bokeh"
+
+    def __init__(
+        self, figure, key: str = None, target_path: str = None,
+    ):
+        """
+        Initialize a Bokeh artifact with the given figure.
+
+        :param figure:      Bokeh figure ('bokeh.plotting.Figure' object) to save as an artifact.
+        :param key:         Key for the artifact to be stored in the database.
+        :param target_path: Path to save the artifact.
+        """
+        super().__init__(key=key, target_path=target_path, viewer="bokeh")
+
+        # Validate input:
+        try:
+            from bokeh.plotting import Figure
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                "Using 'BokehArtifact' requires bokeh>=2.3.2 "
+                "but bokeh is not found: {}".format(e)
```

Comment:
```suggestion
                "Using 'BokehArtifact' requires bokeh package. Use pip install mlrun[bokeh] to install it"
```

Reply:
Also are you sure it shouldn't be `ImportError` here ?

Reply:
![image](https://user-images.githubusercontent.com/83535508/128932265-f483a3f9-f03b-4136-bc7f-cc093bab29a9.png)